### DISCUSSION TITLE
Labels

### DISCUSSION THREAD

**Discussion by aug70** [arendst/Tasmota#18450] [Q&A]
Is it possible to label devices? Or is there an easier way to accomplish the same objective?
I want to be able to discover devices through MQTT based on their labels.

**Comment by barbudor** [comment]
You can identify devices by their topic
Alternatively you can set a DeviceName

Not sure how you want to discover devices
Each device send some message on start which could be retained like LWT, INFO1/2/3
There's also the Tasmota Discovery protocol

**Comment by lalo-uy** [comment]
The devices have:
Device name
Host name
Topic

What you mean by discovering?
On mqtt you may onli see Topic , unless you ast for status 5

**Reply by aug70** [reply]
For example when searching for webcams; `StatusFWR` has a value of `{"Version":"12.4.0(webcam)...` if I want to discover all webcams, I can search/regex based on that field value!
I'm aware of those fields Name, FriendlyName, Hostname etc. But if I want to find another group of devices that are tagged together, e.g. Location, or purpose or brand, or whatever grouping I can think of one possible way of doing this is using `FriendlyName` field and generate names for devices that can serve as labels. But this is not a very clear solution. It'd be pretty restrictive. 
I was looking for a way to add k/v pairs as labels to devices.

**Reply by sfromis** [reply]
You could also have a boot time rule to publish your own retained messages with whatever you want to have there, and thus easily available with just a MQTT subscription.

Another way of saving tags would be in persistent Mem variables, however, these are not available to discovery without a query.

**Reply by aug70** [reply]
@sfromis That sounds interesting, can you point me to a documentation or example?

**Reply by sfromis** [reply]
You can read docs about rules, and the Publish command.

**Comment by sfromis** [comment]
In addition, you can also configure FriendlyName which is included in the Tasmota retained discovery payloads (along with device name, host name and topic).
